File: uniapp-web/src/api/gold.ts
/**
 * 黄金预测 API
 */
import { get } from './request'

export interface GoldPriceInfo {
  id: number
  date: string
  price_cny: number
  price_usd: number
  change_pct: number
  open: number
  high: number
  low: number
  close: number
  volume: number
  created_at?: string
  is_realtime?: boolean
  update_time?: string
  source?: string
}

export interface GoldRealtimeData {
  shanghai?: {
    price: number
    yesterday_close: number
    open: number
    high: number
    low: number
    time: string
    prev_settle: number
    volume: number
    change: number
    change_pct: number
    date: string
    name: string
  }
  newyork?: {
    price: number
    open: number
    high: number
    low: number
    time: string
    prev_settle: number
    date: string
    name: string
  }
  london?: {
    price: number
    prev_close: number
    open: number
    high: number
    low: number
    time: string
    date: string
    name: string
  }
}

export interface GoldPredictionInfo {
  id?: number
  predict_date: string
  predict_price: number
  algorithm: string
  confidence: number
  trend: string
  change_pct: number
  created_at?: string
}

export interface GoldPredictResult {
  current_price: number
  current_date: string
  algorithm: string
  predictions: GoldPredictionInfo[]
  confidence_avg: number
  trend_summary: string
}

export interface GoldPredictParams {
  days_options: { value: number; label: string; desc: string }[]
  algorithm_options: { value: string; label: string; desc: string }[]
}

export function getGoldPrices(days: number = 30) {
  return get<GoldPriceInfo[]>('/api/gold/prices', { days })
}

export function getLatestPrice() {
  return get<GoldPriceInfo>('/api/gold/latest')
}

export function getRealtimeGold() {
  return get<GoldRealtimeData>('/api/gold/realtime')
}

export function predictGold(days: number = 7, algorithm: string = 'ensemble') {
  return get<GoldPredictResult>('/api/gold/predict', { days, algorithm })
}

export function getGoldPredictions(limit: number = 20) {
  return get<GoldPredictionInfo[]>('/api/gold/predictions', { limit })
}

export function getGoldPredictParams() {
  return get<GoldPredictParams>('/api/gold/params')
}

File: uniapp-server/schemas/gold.py
"""黄金预测相关数据模型"""
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel


class GoldPriceInfo(BaseModel):
    """黄金价格信息"""
    id: int
    date: str
    price_cny: float
    price_usd: Optional[float] = 0
    change_pct: Optional[float] = 0
    open: Optional[float] = 0
    high: Optional[float] = 0
    low: Optional[float] = 0
    close: Optional[float] = 0
    volume: Optional[int] = 0
    created_at: Optional[datetime] = None
    is_realtime: Optional[bool] = False
    update_time: Optional[str] = ""
    source: Optional[str] = ""

    class Config:
        from_attributes = True


class GoldRealtimeInfo(BaseModel):
    """实时金价信息"""
    shanghai: Optional[dict] = None
    newyork: Optional[dict] = None
    london: Optional[dict] = None


class GoldPredictRequest(BaseModel):
    """预测请求"""
    days: int = 7
    algorithm: str = "ensemble"


class GoldPredictionInfo(BaseModel):
    """预测结果信息"""
    id: Optional[int] = None
    predict_date: str
    predict_price: float
    algorithm: str = "ensemble"
    confidence: float = 0
    trend: str = "stable"
    change_pct: float = 0
    created_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class GoldPredictResult(BaseModel):
    """完整预测结果"""
    current_price: float
    current_date: str
    algorithm: str
    predictions: List[GoldPredictionInfo]
    confidence_avg: float = 0
    trend_summary: str = "stable"


class GoldPredictParams(BaseModel):
    """预测参数选项"""
    days_options: List[dict] = []
    algorithm_options: List[dict] = []
